mu-plugins/blocks/time/src/convert_times.js

```diff
@@ -0,0 +1,80 @@
+function convertTimes() {
+	const parseDate = function ( text ) {
+		const match =
+			/^([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2}):([0-9]{2})\+([0-2]{1})([0-9]{1}):00$/.exec(
+				text
+			);
+
+		return (
+			match &&
+			new Date(
+				// Date.UTC(year, monthIndex (0..11), day, hour, minute, second)
+				Date.UTC( +match[ 1 ], +match[ 2 ] - 1, +match[ 3 ], +match[ 4 ], +match[ 5 ], +match[ 6 ] )
+			)
+		);
+	};
+
+	const formatTime = function ( date ) {
+		return date.toLocaleTimeString( window.navigator.language, {
+			weekday: 'long',
+			month: 'long',
+			day: 'numeric',
+			year: 'numeric',
+			hour: '2-digit',
+			minute: '2-digit',
+			timeZoneName: 'short',
+		} );
+	};
+
+	const formatDate = function ( date ) {
+		return date.toLocaleDateString( window.navigator.language, {
+			weekday: 'long',
+			month: 'long',
+			day: 'numeric',
+			year: 'numeric',
+		} );
+	};
+
+	// Not all browsers, particularly Safari, support arguments to .toLocaleTimeString().
+	const toLocaleTimeStringSupportsLocales = ( function () {
+		try {
+			new Date().toLocaleTimeString( 'i' );
+		} catch ( event ) {
+			return event.name === 'RangeError';
+		}
+
+		return false;
+	} )();
+
+	document.querySelectorAll( '.wporg-time' ).forEach( ( dateElement ) => {
+		let localTime = '';
+		const datetime = dateElement.getAttribute( 'datetime' );
+		const datetimeISO = dateElement.getAttribute( 'data-iso' );
+		const date = datetime && parseDate( datetime );
```

Comment:
This was coming up null for me with a valid time: 2023-07-12T19:00:00-04:00

Although, I think that timestamp is wrong, it should either be `2023-07-12T19:00:00-00:00` or `2023-07-12T15:00:00-04:00`

Reply:
Looks like the regex didn't allow negative offsets, which didn't affect me with `+12`. Can you please try again?

Reply:
I've changed my mac to `America/New_York` and with the date string 'Wednesday July 12 2023 at 15:00 EDT' I now see

![Screenshot 2023-07-24 at 9 25 44 PM](https://github.com/WordPress/wporg-mu-plugins/assets/1017872/29a76161-f4a0-4117-a50f-7f1ca5f46826)

While with my local timezone I see

![Screenshot 2023-07-25 at 11 52 22 AM](https://github.com/WordPress/wporg-mu-plugins/assets/1017872/f92c822b-71b6-4955-a2d2-2582095f32be)

Reply:
This seems consistent with the shortcode version on this draft https://make.wordpress.org/core/?p=106234&preview=true